### PR TITLE
feat: add config validation

### PR DESCRIPTION
**Pull request by csatib02**
## Overview

- This PR adds the follow-up config validation of #92.

1. The assembled config is converted into the internally used config, via `mapstructure` and `decode-hooks`.
2. Then the imported `otel-collector` config validator will handle the heavy-lifting.

### CODE REVIEW

File: internal/controller/telemetry/collector_controller.go [hunk 1]
```diff
@@ -199,16 +200,19 @@ func (r *CollectorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{}, err
 	}
 
-	// NOTE: This might be revised or removed in the future, but good enough for now to avoid
-	// deploying the collector that would immediately error due to configuration errors.
-	// Might also be a good place to add a validation webhook to validate the collector spec
-	if err := otelConfigInput.ValidateConfig(); err != nil {
+	// NOTE: Might also be a good place to add a validation webhook to validate the config input
```
Comment (pepov) [inline internal/controller/telemetry/collector_controller.go:203]: This isn't relevant here I think
```suggestion
```

**Review by pepov — APPROVED** [review]
:godmode:

File: internal/controller/telemetry/collector_controller.go (hunk 1 above)
Comment (pepov) [reply internal/controller/telemetry/collector_controller.go:203]: However the idea is right. We need a webhook to make the ux of the validation more user and operator friendly.

Comment (pepov) [reply internal/controller/telemetry/collector_controller.go:203]: And that should be a follow up task, not in scope right now.

**Review by OverOrion — APPROVED** [review]
LGTM, good job! :rocket:

**Pull request merged**